File: bo/types/member.ts
export interface Club {
  id: string; // UUID
  name: string;
  email?: string;
  phone?: string;
  address?: string;
  city?: string;
  website?: string;
  creationDate?: string; // ISO date
  description?: string;
  logoUrl?: string;
  status?: string; // 'actif' | 'inactif'
  // Bureau du club
  president?: string;
  vicePresident?: string;
  tresorier?: string;
  secretaire?: string;
}

export enum ClubRole {
  PRESIDENT = "PRESIDENT",
  VICE_PRESIDENT = "VICE_PRESIDENT",
  SECRETAIRE = "SECRETAIRE",
  TRESORIER = "TRESORIER",
  MEMBRE = "MEMBRE",
  ENTRAINEUR = "ENTRAINEUR",
  ARBITRE = "ARBITRE",
}

export interface Member {
  id: string; // UUID
  name: string; // pseudo
  email: string;
  phone?: string;
  address?: string;
  city?: string;
  rate?: number;
  role?: string; // rôle système (ROLE_USER, ROLE_ADMIN)
  // Champs profil
  firstName?: string;
  lastName?: string;
  birthDate?: string; // ISO string (YYYY-MM-DD)
  gender?: string; // 'M' | 'F' | 'Autre'
  active?: boolean;
  // Champs supplémentaires pour profil complet
  registrationDate?: string; // Date d'inscription (ISO string)
  currentPoints?: number; // Capital points actuel
  licenceNumber?: string; // Numéro de licence
  ffjdId?: string; // ID FFJD du joueur
  ranking?: number; // Classement ELO
  // Relations club
  club?: Pick<Club, "id"> | null; // Relation backend : objet avec id uniquement
  clubId?: number; // ID du club (pour formulaires)
  clubName?: string; // nom du club (pour affichage, retourné par backend)
  clubRole?: ClubRole; // rôle dans le club
}

// Alias éventuel si l'on souhaite distinguer plus tard
export type User = Member;

File: api/src/main/java/com/damier/damierclub/model/Member.java
package com.damier.damierclub.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.damier.damierclub.util.UuidGenerator;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "members")
public class Member {

    @Id
    @Column(columnDefinition = "uuid")
    private UUID id;

    @NotBlank
    private String name; // pseudo
    @Email
    @Column(unique = true)
    private String email;
    private String phone;
    private String address;
    private String city;
    private int rate;
    @JsonIgnore
    private String password;

    // Champs profil
    private String firstName;
    private String lastName;
    private LocalDate birthDate;
    private String gender;
    private Boolean active = true;

    // Champs supplémentaires pour le profil complet
    private LocalDate registrationDate; // Date d'inscription
    private Integer currentPoints = 0; // Capital points actuel
    private String licenceNumber; // Numéro de licence

    // Relation ManyToOne : un membre appartient à un club
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "club_id")
    private Club club;

    // Rôle du membre dans le club (ex: PRESIDENT, SECRETAIRE, TRESORIER, MEMBRE)
    @Enumerated(EnumType.STRING)
    private ClubRole clubRole = ClubRole.MEMBRE;

    // Rôle système (ex: ADMIN, USER)
    private String role = "ROLE_USER";

    @PrePersist
    public void generateId() {
        if (this.id == null) {
            this.id = UuidGenerator.generateUuidV7();
        }
    }
}
